src/clr-angular/data/datagrid/providers/items.ts

```diff
@@ -178,7 +178,8 @@ export class Items<T = any> {
    * Extracts the current page from the sorted list
    */
   private _changePage() {
-    if (this.uninitialized) {
+    // If we know we have pagination but the page size hasn't been set yet, we wait for it.
+    if (this.uninitialized || (this._page.activated && this._page.size === 0)) {
```

Comment:
Do we have a test to cover this? I haven't looked recently, but this is a new code path that we could extend an existing test to validate.